### PR TITLE
Fix MSVC non-member template functions compilation error

### PR DESCRIPTION
**Pull request by suVrik**
Hey! With a recent MSVC update my project fails to compile because I use non-member template functions in EnTT reflection. For whatever reason MSVC considers these functions to be references rather than pointers. EnTT tests fail to compile as well, even though non-member template `func` is not covered (non-member template setters and getters are). Here's the fix for this issue. I added tests for both member template functions and non-member ones just in case of future MSVC surprises. Cheers

### CODE REVIEW

**Comment by skypjack** [comment]
Good catch, thank you for pointing this out. May I ask you what version of MSVC is it?
Feel free also to add you nick to the list of contributors if you want (file AUTHORS, lexicographic order).

**Comment by suVrik** [comment]
I'm already on the list! My MSVC version is 2019 16.4.1. Compiler version is 14.24.28314.

**Comment by skypjack** [comment]
Oh, sorry, I'm from mobile and I didn't check before writing the message. I apologize.

**Comment by skypjack** [comment]
Since the problem is on the current `master` and also on the last tag (see #368), I've decided to cut a new patch release of `EnTT`.
However, this PR has some conflicts with the work done on `experimental`, that is the upcoming `master` (hopefully by the end of the month).

My plan is to port only the relevant changes to the `.hpp` files on `master` so as to reduce the conflicts, then add the tests directly on the rework from `experimental`.
Put simply, the new tests strengthen the test suite but they aren't strictly necessary for the patch since it's easily reproducible with the current tests. So we can safely split the whole thing in two parts. :+1:
